### PR TITLE
R 4.0 tests fixes

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

File: h2o-r/tests/testdir_algos/gbm/runit_GBM_quantile.R [hunk 1]
```diff
@@ -64,7 +64,8 @@ gbm.quantile.test = function(){
   #a test to look at h2o.quantile() - init_F is equal to POJO splits for one node.
   left_q = h2o.quantile(data.hex.test.left[,resp],Q) 
   left_q_test = left_q - m@model$init_f
-  print(paste("Quantile GBM left split", left_q_test, LEFT_POJO))
+  left_q_test
+  LEFT_POJO
```
Comment (sebhrusen) [inline h2o-r/tests/testdir_algos/gbm/runit_GBM_quantile.R]: this does nothing, does it?

Comment (honzasterba) [reply h2o-r/tests/testdir_algos/gbm/runit_GBM_quantile.R]: true, eventually I ended up not touching this test

File: h2o-r/tests/runitUtils/utilsR.R [hunk 2]
```diff
@@ -409,6 +410,18 @@ function(..., envir=parent.frame()) {
     list(tests=tests, envir=envir)
 }
 
+setSeedPriorR36<-
+function(seed) {
+    if (as.numeric(R.Version()$major) >= 4 || 
+        (as.numeric(R.Version()$major) == 3 && as.numeric(substr(R.Version()$minor, 1, 1)) >= 6)
```
Comment (sebhrusen) [inline h2o-r/tests/runitUtils/utilsR.R:416]: strictly speaking, to recover the minor version (imagine 3.11.5), you should use sth like
```
as.numeric(unlist(strsplit(R.Version()$minor, '.', fixed=TRUE))[[1]])
```

File: h2o-r/tests/testdir_misc/runit_quantile.R [hunk 3]
```diff
@@ -1,46 +1,44 @@
...
-aa$h_wts = as.h2o(wts)
-#r_q = wtd.quantile(x, wts, probs = seq(0,1,.05))
-r_q=c(3.895489e-06,4.863379e-02,9.789691e-02,1.470487e-01,1.977443e-01,2.473365e-01,2.975013e-01,3.482667e-01,3.980460e-01,4.483631e-01,4.990024e-01,5.489128e-01,5.986945e-01,6.486255e-01,6.991498e-01,7.500031e-01,8.001472e-01,8.504057e-01,8.996923e-01,9.498159e-01,9.999471e-01)
-h_q = h2o.quantile(aa,probs = seq(0,1,.05),weights_column = "h_wts")
-expect_true(max(abs((r_q-h_q)/r_q)) < 1e-5)
+#This tests quantile and weighted quantile on synthetic data by comparing with R
+test.quantile <- function() {
 
+    # set random seed to generate random dataset
+    set.seed(1234)
```
Comment (sebhrusen) [inline h2o-r/tests/testdir_misc/runit_quantile.R:8]: so there are cases for which `set.seed` is reproducible accross all versions, and some for which we should rely on `setSeedPriorR36`?
How do we know? Shouldn't we make `setSeedPriorR36` our default?
```
setSeed <- set.seed
set.seed <- setSeedPriorR36  # use setSeed internally
```

Comment (honzasterba) [reply h2o-r/tests/testdir_misc/runit_quantile.R:8]: good idea, will add

File: h2o-r/tests/runitUtils/utilsR.R (hunk 2 above)
Comment (honzasterba) [reply h2o-r/tests/runitUtils/utilsR.R:416]: fixed

**Pull request merged**